composer/utils/checkpoint.py

```diff
@@ -1107,12 +1109,13 @@ def _save_checkpoint(
 
         log.debug(f'Saving sharded checkpoints to {save_filename}...')
         process_group = None
-        device_mesh = state.fsdp_device_mesh
+        device_mesh = state.device_mesh
         if device_mesh is not None and device_mesh.ndim == 2:
             # If hybrid shard, only rank in first replica saves
             expect_file = device_mesh.get_local_rank(mesh_dim=0) == 0
             if expect_file:
                 process_group = device_mesh.get_group(1)  # Shard process_group for first replica
+                assert isinstance(process_group, ProcessGroup)
```

Comment:
is this simply for linting or is there another reason

Reply:
Will add comment, yes for typechecker